eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqliteAuditDb.scala

```diff
@@ -37,34 +37,63 @@ class SqliteAuditDb(sqlite: Connection) extends AuditDb with Logging {
+      statement.executeUpdate("CREATE TABLE sent (amount_msat INTEGER NOT NULL, fees_msat INTEGER NOT NULL, final_amount_msat INTEGER NOT NULL, payment_id TEXT NOT NULL, parent_payment_id TEXT NOT NULL, payment_hash BLOB NOT NULL, payment_preimage BLOB NOT NULL, recipient_node_id BLOB NOT NULL, to_channel_id BLOB NOT NULL, timestamp INTEGER NOT NULL)")
+      // Old rows will be missing a recipient node id, so we use an easy-to-spot default value.
+      val defaultRecipientNodeId = PrivateKey(ByteVector32.One).publicKey
+      statement.executeUpdate(s"INSERT INTO sent (amount_msat, fees_msat, final_amount_msat, payment_id, parent_payment_id, payment_hash, payment_preimage, recipient_node_id, to_channel_id, timestamp) SELECT amount_msat, fees_msat, amount_msat, id, id, payment_hash, payment_preimage, X'${defaultRecipientNodeId.toString}', to_channel_id, timestamp FROM _sent_old")
+      statement.executeUpdate("DROP table _sent_old")
+
+      statement.executeUpdate("DROP INDEX relayed_timestamp_idx")
+      statement.executeUpdate("ALTER TABLE relayed RENAME TO _relayed_old")
+      statement.executeUpdate("CREATE TABLE relayed (payment_hash BLOB NOT NULL, amount_msat INTEGER NOT NULL, channel_id BLOB NOT NULL, direction TEXT NOT NULL, relay_type TEXT NOT NULL, timestamp INTEGER NOT NULL)")
+      statement.executeUpdate("INSERT INTO relayed (payment_hash, amount_msat, channel_id, direction, relay_type, timestamp) SELECT payment_hash, amount_in_msat, from_channel_id, 'IN', 'channel', timestamp FROM _relayed_old")
+      statement.executeUpdate("INSERT INTO relayed (payment_hash, amount_msat, channel_id, direction, relay_type, timestamp) SELECT payment_hash, amount_out_msat, to_channel_id, 'OUT', 'channel', timestamp FROM _relayed_old")
+      statement.executeUpdate("DROP table _relayed_old")
+
+      statement.executeUpdate("CREATE INDEX IF NOT EXISTS sent_timestamp_idx ON sent(timestamp)")
+      statement.executeUpdate("CREATE INDEX IF NOT EXISTS relayed_timestamp_idx ON relayed(timestamp)")
+      statement.executeUpdate("CREATE INDEX IF NOT EXISTS relayed_payment_hash_idx ON relayed(payment_hash)")
+    }
+
     getVersion(statement, DB_NAME, CURRENT_VERSION) match {
       case 1 => // previous version let's migrate
         logger.warn(s"migrating db $DB_NAME, found version=1 current=$CURRENT_VERSION")
         migration12(statement)
         migration23(statement)
+        migration34(statement)
         setVersion(statement, DB_NAME, CURRENT_VERSION)
       case 2 =>
         logger.warn(s"migrating db $DB_NAME, found version=2 current=$CURRENT_VERSION")
         migration23(statement)
+        migration34(statement)
+        setVersion(statement, DB_NAME, CURRENT_VERSION)
+      case 3 =>
+        logger.warn(s"migrating db $DB_NAME, found version=3 current=$CURRENT_VERSION")
+        migration34(statement)
         setVersion(statement, DB_NAME, CURRENT_VERSION)
       case CURRENT_VERSION =>
```

Comment:
How about using triple quotes for this kind of queries?

Example:

```scala
statement.executeUpdate("""
  CREATE TABLE IF NOT EXISTS sent (
    amount_msat INTEGER NOT NULL, 
    fees_msat INTEGER NOT NULL, 
    final_amount_msat INTEGER NOT NULL,
    payment_id TEXT NOT NULL, -- renamed v4
    parent_payment_id TEXT NOT NULL, 
    payment_hash BLOB NOT NULL, 
    payment_preimage BLOB NOT NULL, 
    recipient_node_id BLOB NOT NULL,
    to_channel_id BLOB NOT NULL, 
    timestamp INTEGER NOT NULL
)""")
```

Of course it would make the database files feel more verbose, but it would also help understand what a table looks like right now, and how it has changed when doing a git `diff`.

Reply:
FWIW, having it all in one line allows quick checking that the table declaration in the migration function is identical to the one in the "regular" path.

Reply:
I don't have a strong opinion on that.
The way I saw it was that the code in the `CURRENT_VERSION` didn't need to mention what changed: in most cases you're only interested in what the schema is now.
If you're interested in what the schema was before, then you look at migration code and tests.